source/server/connection_handler_impl.cc

```diff
@@ -83,16 +81,27 @@ void ConnectionHandlerImpl::addListener(absl::optional<uint64_t> overridden_list
   if (auto* listener = details.listener_->listener(); listener != nullptr) {
     listener->setRejectFraction(listener_reject_fraction_);
   }
-  listeners_.emplace_back(config.listenSocketFactory().localAddress(), std::move(details));
+
+  ASSERT(!listener_map_by_tag_.contains(config.listenerTag()));
+
+  listener_map_by_tag_.emplace(
+      config.listenerTag(),
+      std::make_pair(config.listenSocketFactory().localAddress(), std::move(details)));
+  // This map only store the new listener.
+  listener_map_by_address_.insert_or_assign(config.listenSocketFactory().localAddress()->asString(),
+                                            &listener_map_by_tag_[config.listenerTag()].second);
```

Comment:
Can `absl::flat_hash_map` guarantee the stability of value address? May we should use `absl::node_hash_map` for `listener_map_by_tag_` or wrapper the details with `unique_ptr`? 🤔

Reply:
Oh, right, absl::flat_hash_map not guarantee that. thanks for point that out! I can use the unique_ptr, or std::reference_wrapper.

Reply:
@rojkov in the end, I use flat_hash_map + unique_ptr, let me know if you still think `node_hash_map` is a better option.